books-refractored.js

```diff
@@ -0,0 +1,675 @@
+    booksToDisplay = filteredBooks;
+  } else if (isGenreFiltered && !isYearFiltered) {
+    booksToDisplay = filteredBooks;
+  } else if (!isGenreFiltered && isYearFiltered) {
+    booksToDisplay = filteredBooks;
+  } else if (!isGenreFiltered && !isYearFiltered) {
+    console.log("stops at all");
+    booksToDisplay = books;
+  } else if (!(previousFilter === null) && isGenreFiltered && isYearFiltered) {
+    booksToDisplay = genreFilteredBooks.filter((book) =>
+      yearFilteredBooks.includes(book)
+    );
+  } else {
+    console.log("something is wrong");
+  } // get the intersection of filteredBooks array and newFilterBooks
+
+  booksToDisplay.length === 0
+    ? console.log("no result")
+    : console.log("there are books left");
+
+  previousFilter = filterType;
+};
+
+//Apply sorting
+//Part 1: Display books
+const applySorting = () => {
+  if (!(booksToDisplay.length === 0)) {
+    if (sortingFunction) {
+      //Check if there is any sorting applied
+      booksToDisplay = sortingFunction(booksToDisplay);
+    }
+
+    bookCards(booksToDisplay);
+  } else {
+    bookDisplay.innerHTML = `<p class="no-result">No result.</p>`;
+  }
+};
+
+//Part 2: Toogle Style
+//----click once to select, click again to deselect
+const checkSorted = (sortFunc) => {
+  if (isSorted && previousSort === sortFunc) {
+    //Deselect sorting
+    sortingFunction = null;
+    isSorted = false;
+  } else {
+    //Select sorting
+    sortingFunction = sortFunc;
+    isSorted = true;
+  }
+
+  applySorting();
+  previousSort = sortingFunction;
+};
+
+const sortByHighRating = (arr) => {
+  const arrCopy = arr.slice();
+  sortedBooks = arrCopy.sort((a, b) => b.rating - a.rating);
+  return sortedBooks;
+};
+
+const sortByPublish = (arr) => {
+  const arrCopy = arr.slice();
+  sortedBooks = arrCopy.sort((a, b) => b.year - a.year);
+  return sortedBooks;
+};
+
+const sortByAuthor = (arr) => {
+  const arrCopy = arr.slice();
+  sortedBooks = arrCopy.sort((a, b) => {
+    if (a.author < b.author) {
+      return -1;
+    } else if (a.author > b.author) {
+      return 1;
+    } else {
+      return 0;
+    }
+  });
+  console.log(sortedBooks);
+  return sortedBooks;
+};
+
+const sortByTileAZ = (arr) => {
+  const arrCopy = arr.slice();
+  sortedBooks = arrCopy.sort((a, b) => {
+    const stringA = String(a.title);
+    const stringB = String(b.title);
+
+    if (stringA < stringB) {
+      return -1;
+    } else if (stringA > stringB) {
+      return 1;
+    } else {
+      return 0;
+    }
+  });
+  console.log(sortedBooks);
+  return sortedBooks;
+};
+
+const sortByTileZA = (arr) => {
+  const arrCopy = arr.slice();
+  sortedBooks = arrCopy.sort((a, b) => {
+    const stringA = String(a.title);
+    const stringB = String(b.title);
+
+    if (stringA < stringB) {
+      return 1;
+    } else if (stringA > stringB) {
+      return -1;
+    } else {
+      return 0;
+    }
+  });
+  console.log(sortedBooks);
```

Comment:
I see you have `console.log` statements in parts of the code. They are not needed when project is done. ✨